ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,104 @@
+
+A Workspace is an abstraction of a namespace. Each workspace has a single namespace
+which backs it.
+
+Workspaces are required by [kubesaw] for providing a transparent multi-cluster communication for us. A user's workspace may be allocated to one cluster or another - but our proxy layer keeps the user blissfully unaware of this. Workspaces can be placed on clusters either automatically (using [kubesaw] mechanisms) or manually by a platform administrator.
+
+Workspace roles give a way to describe who has what rights in the workspace.
+
+Workspaces happen to be our mechanism for managing quota - a fact inherited from the [kubesaw] stack.
+
+### Some changes
+
+Some relevant changes that have occurred or are ongoing.
+
+* We want to make Konflux a successful open source projects that fits into the
+cloud native apps ecosystem. We want everything to work on vanilla Kubernetes - no OpenShift required.
+
+* We had our [first upstream Konflux release](https://github.com/konflux-ci/konflux-ci) which doesn't use [kubesaw] or relying
+on the Workspace abstraction, and is deployable on vanilla Kubernetes.
+
+* We made a decision (now long ago) to drop [KCP] from our architecture, which initially introduced the workspace concept. Both transparent multi-cluster as a feature and workspaces as a concept came from KCP.
+
+* We made the decision (at Red Hat) to stop trying to offer a managed service interface for Konflux. The user signup, waitlist, and workspace provisioning interfaces made a lot of sense there.
+
+## Decision
+
+1. We will remove the Workspace abstraction from Konflux and use standard Kubernetes namespaces
+directly for providing a place for users to run their pipelines and store their configurations
+and secrets.
+
+2. Konflux won't provide its own policy engine for enforcing policies related to
+creating/deleting namespaces. It will instead recommend user to use existing open source projects
+such as [Kyverno](https://kyverno.io/docs/introduction/) and [Gatekeeper](https://open-policy-agent.github.io/gatekeeper/website/docs/) for this task.
+
+3. We will stop using the term Workspace, and start to use the term Namespace.
+
+4. The Konflux UI will expose a wizard for creating a new namespace. This
+wizard will be available to any user that has permissions to create namespaces.
+
+5. Konflux will provide an optional automation for creating a namespace for a user
+when he/she logins for the first time.
+
+6. Konflux won't be opinionated about the mechanism for initializing/maintaining namespaces with
+supporting resources such as ResourceQuota and LimitRange. Konflux will defer
+this responsibility to other tools which are specialized in this task such as 
+(but not limited to) [Kyverno](https://kyverno.io/policies/best-practices/add-ns-quota/add-ns-quota/) and [ArgoCD](https://github.com/konflux-ci/namespace-generator).
```

Comment:
It seems to me that we are still opinionated on namespace provisioning and management, though. If we don't want to cope with this, I'd suggest the following changes

```suggestion
6. Konflux won't be opinionated about the mechanism for initializing/maintaining namespaces with
supporting resources such as ResourceQuota and LimitRange. Konflux will defer
this responsibility to other tools which are specialized in this task such as 
(but not limited to) [Kyverno](https://kyverno.io/policies/best-practices/add-ns-quota/add-ns-quota/) and [ArgoCD](https://github.com/konflux-ci/namespace-generator).
```

Reply:
@filariow are you suggesting that the Konflux UI won't expose a button for creating a namespace?

Reply:
nope, if I have to suggest something I'd suggest having a `Create Workspace` button 😄

I'm just noticing that it seems to me that the request compute power/namespace/workspace use case still seems important to Konflux

Reply:
If the are no objections I'll close this thread.

Reply:
I don't think this ADR should include a specific requirement for having a workspace/namespace creating button in the UI, as that brings us right back into the WS/NS management business that we are cliaming we want to be out of.

Reply:
> I don't think this ADR should include a specific requirement for having a workspace/namespace creating button in the UI, as that brings us right back into the WS/NS management business that we are cliaming we want to be out of.

@ralphbean @brianwcook what do you think? is it a hard requirement to have a button for creating namespaces in the UI?